### ISSUE TITLE
error reading diff in tagPolicy.gitCommit

### ISSUE THREAD

**Issue by ahmetb** [GoogleContainerTools/skaffold#778]
skaffold v0.8.0

I have a skaffold.yaml like this

```
apiVersion: skaffold/v1alpha2
kind: Config
build:
  tagPolicy:
    gitCommit: {}
```

I just created a `src/loadgenerator` directory (but didn't do `git add` yet), and now I added this to my `artifacts` section:

```
  artifacts:
  ....
  - imageName: frontend
    workspace: src/frontend
  - imageName: loadgenerator
    workspace: src/loadgenerator
```

I'm getting this error:

> `FATA[0001] first build: exiting dev mode because the first build failed: generating tag: reading diff: read /Users/ahmetb/workspace/myproject/src/loadgenerator: is a directory`

It looks like a git error.

### Verbose output

```
[0001] Running command: [git rev-parse --show-toplevel]
DEBU[0001] Command output: stdout /Users/ahmetb/workspace/myproject
, stderr:
DEBU[0001] Running command: [git rev-parse HEAD]
DEBU[0001] Command output: stdout f1920e2ca3dec6ccfea381cba1ee88bc8dd80652
, stderr:
DEBU[0001] Running command: [git status --porcelain]
DEBU[0001] Command output: stdout  M skaffold.yaml
 M src/shippingservice/main.go
?? kubernetes-manifests/loadgenerator.yaml
?? src/loadgenerator/
, stderr:
FATA[0001] first build: exiting dev mode because the first build failed: generating tag: reading diff: read /Users/ahmetb/workspace/myproject/src/loadgenerator: is a directory
```

**Comment by ahmetb** [comment]
I just got another error after `git add`:

> `FATA[0001] first build: exiting dev mode because the first build failed: generating tag: reading diff: open /Users/ahmetb/workspace/myproject/ kubernetes-manifests/loadgenerator.yaml: no such file or directory`

Note the space between `myproject/ kubernetes-manifests`. The file exists, but without that space.

**Comment by dgageot** [comment]
@ahmetb This should be fixed in 0.9.0
